datalad/interface/common_cfg.py

```diff
@@ -28,9 +29,185 @@
 )
 from datalad.utils import on_windows
 
+lgr = logging.getLogger('datalad.interface.common_cfg')
 dirs = AppDirs("datalad", "datalad.org")
 
 
+class _NotGiven():
+    pass
+
+
+def register_config(
+        name,
+        title,
+        *,
+        default=_NotGiven,
+        default_fn=_NotGiven,
+        description=None,
+        # yes, we shadow type, this is OK
+        type=_NotGiven,
+        # for manual entry
+        dialog=None,
+        scope=_NotGiven,
+    ):
+    """Register a configuration item
+
+    This function can be used by DataLad extensions and other client
+    code to register configurations items and their documentation with
+    DataLad's configuration management. Specifically, these definitions
+    will be interpreted by and acted on by the `configuration` command,
+    and `ConfigManager.obtain()`.
+
+    At minimum, each item must be given a name, and a title. Optionally, any
+    configuration item can be given a default (or a callable to compute a
+    default lazily on access), a type-defining/validating callable (i.e.
+    `Constraint`), a (longer) description, a dialog type to enable manual
+    entry, and a configuration scope to store entered values in.
+
+    Parameters
+    ----------
+    name: str
+      Configuration item name, in most cases starting with the prefix
+      'datalad.' followed by at least a section name, and a variable
+      name, e.g. 'datalad.section.variable', following Git's syntax for
+      configuration items.
+    title: str
+      The briefest summary of the configuration item's purpose, typically
+      written in the style of a headline for a dialog UI, or that of an
+      explanatory inline comment just prior the item definitions.
+    default: optional
+      A default value that is already known at the time of registering the
+      configuration items. Can be of any type.
+    default_fn: callable, optional
+      A callable to compute a default value lazily on access. The can be
+      used, if the actual value is not yet known at the time of registering
+      the configuration item, or if the default is expensive to compute
+      and its evaluation need to be deferred to prevent slow startup
+      (configuration items are typically define as one of the first things
```

Comment:
```suggestion
      and its evaluation needs to be deferred to prevent slow startup
      (configuration items are typically defined as one of the first things
```

Reply:
Thanks! I will apply this patch in the next update.